File: src/modularization_programs/c/generic_example/sample_functions.c
/*
 * Author: Jhonatan Casale (jhc)
 *
 * Contact : jhonatan@jhonatancasale.com
 *         : casale.jhon@gmail.com
 *         : https://github.com/jhonatancasale
 *         : https://twitter.com/jhonatancasale
 *         : http://jhonatancasale.github.io/
 *
 * Create date Wed 23 Nov 03:39:50 BRST 2016
 *
 */

void function_receive_and_return_void ( void );
void function_receive_something_return_void ( int, int );
int function_receive_void_return_something ( void );
int function_receive_and_return_something ( int, int );

#include <stdlib.h>
#include <stdio.h>

int main (int argc, char **argv)
{
  function_receive_and_return_void ();
  function_receive_something_return_void ( 3, 4 );
  printf ( "Random trash: %d\n", function_receive_void_return_something () );
  printf ( "(a + b) = %d\n", function_receive_and_return_something ( 10, 7 ) );


  return (EXIT_SUCCESS);
}

void function_receive_and_return_void ( void ){ 
  printf ( "void -> void\n" ); 
}

void function_receive_something_return_void ( int a, int b ){ 
  printf ( "(a + b) = %d\n", a + b ); 
}

int function_receive_void_return_something ( void ){ 
  int i;
  return i;
}

int function_receive_and_return_something ( int a, int b ){ 
  return ( a + b ); 
}

File: src/modularization_programs/python/generic_example/sample_functions.py
"""
  Author: Jhonatan Casale (jhc)
 
  Contact : jhonatan@jhonatancasale.com
          : casale.jhon@gmail.com
          : https://github.com/jhonatancasale
          : https://twitter.com/jhonatancasale
          : http://jhonatancasale.github.io/
 
  Create date Fri 25 Nov 20:16:41 BRST 2016
"""

import random

def function_receive_and_return_void ():
    '''(void) -> void

    Print a dummy message
    '''

    print('void -> void')

def function_receive_something_return_void (a, b):
    '''(number, number) -> void 

    Print a plus b
    '''

    print('(a + b) = ({} + {}) = {}'.format(a, b, a+b) )

def function_receive_void_return_something ():
    '''(void) -> float

    Return a random number between 0 and 1
    '''

    return random.random()

def function_receive_and_return_something (a, b):
    '''(number, number) -> number

    Return: a + b
    '''

    return a + b

i, j = random.choice( range( 0, 101 ) ), random.choice( range( 0, 101 ) )

function_receive_and_return_void()
function_receive_something_return_void( i, j )
print( 'Rand: %f' % (function_receive_void_return_something () ) )
print( '(a + b) = ({} + {}) = {}'.format( i, j, 
    function_receive_and_return_something( i, j )) )
